db_sql.go

```diff
@@ -67,12 +67,15 @@ func (x *sqlUserStoreDB) Authenticate(identity, password string) error {
 
 func (x *sqlUserStoreDB) SetPassword(userId UserId, password string) error {
 	if tx, etx := x.db.Begin(); etx == nil {
-		if eupdate := x.setPasswordInternal(tx, userId, password); eupdate == nil {
-			return tx.Commit()
-		} else {
+		if used := x.verifyPasswordArchived(userId, password); used {
```

Comment:
No need for the `used` variable here. Just write `if x.hasPasswordBeenUsedBefore()`